### PR TITLE
Implement various Throne of Eldraine cards

### PR DESCRIPTION
**Pull request by jmharmon**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/g/GoldenEgg.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new DrawCardSourceControllerEffect(1)));
+
+        // {1}, {T}: Sacrifice Golden Egg: Add one mana of any color.
+        Ability ability = new AnyColorManaAbility(new GenericManaCost(1));
+        ability.addCost(new TapSourceCost());
+        ability.addCost(new SacrificeSourceCost());
+        this.addAbility(ability);
+
+        // {2}, {T}, Sacrifice Golden Egg: You gain 3 life.
+        Ability ability1 = new SimpleActivatedAbility(Zone.BATTLEFIELD, new GainLifeEffect(3), new ManaCostsImpl("{1}"));
```
Comment (emerald000) [inline Mage.Sets/src/mage/cards/g/GoldenEgg.java]: Costs 2.

File: Mage.Sets/src/mage/cards/w/WishfulMerfolk.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(2);
+
+        // Defender
+        this.addAbility(DefenderAbility.getInstance());
+
+        // {1}{U}: Wishful Merfolk loses defender and becomes a Human until end of turn.
+        //this.addAbility(new SimpleActivatedAbility(Zone.BATTLEFIELD, new WishfulMerfolkEffect(), new ManaCostsImpl("{1}{U}")));
+       this.addAbility(new SimpleActivatedAbility(Zone.BATTLEFIELD, new BecomesCreatureSourceEffect(new WishfulMerfolkToken(), "", Duration.EndOfTurn), new ManaCostsImpl("{1}{U}")));
```
Comment (emerald000) [inline Mage.Sets/src/mage/cards/w/WishfulMerfolk.java]: This should not turn Wishful Merfolk in a creature if it wasn't one, change the Power/Toughness, ...

Better split the ability in 2 parts (lose defender, and become human).

File: Mage.Sets/src/mage/cards/g/GoldenEgg.java (hunk 1 above)
Comment (jmharmon) [reply Mage.Sets/src/mage/cards/g/GoldenEgg.java]: Fixed.

File: Mage/src/main/java/mage/game/permanent/token/FoodToken.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    public FoodToken(String setCode, int tokenType) {
+        super("Food", "Food token");
+        availableImageSetCodes = tokenImageSets;
+        setOriginalExpansionSetCode(setCode);
+        cardType.add(CardType.ARTIFACT);
+        subtype.add(SubType.FOOD);
+
+        // {2}, {T}, Sacrifice this artifact: You gain 3 life.”
+        Ability ability = new SimpleActivatedAbility(Zone.BATTLEFIELD, new GainLifeEffect(3), new GenericManaCost(2));
+        SacrificeSourceCost cost = new SacrificeSourceCost();
```
Comment (emerald000) [inline Mage/src/main/java/mage/game/permanent/token/FoodToken.java]: Needs a {T} cost.

Comment (jmharmon) [reply Mage/src/main/java/mage/game/permanent/token/FoodToken.java]: Should be better now.

**Pull request merged**